src/Analyzer.Cli.FunctionalTests/CommandLineParserTests.cs

```diff
@@ -122,5 +123,51 @@ public void AnalyzeDirectory_DirectoryWithOtherJsonFiles_LogsExpectedErrorInSari
             Assert.AreNotEqual(null, toolNotifications[1]["exception"]);
             Assert.AreEqual(null, toolNotifications[2]["exception"]);
         }
+
+        [TestMethod]
+        public void AnalyzeDirectory_ExecutionWithErrorAndWarningAndNoVerboseMode_PrintsExpectedLogSummary()
+        {
+            var directoryToAnalyze = GetFilePath("ToTestSummaryLogger");
+
+            var expectedLogSummary = "2 error(s) and 1 warning(s) were found during the execution, please refer to the original messages above" +
+                $"{Environment.NewLine}The verbose mode (option -v or --verbose) can be used to obtain even more information about the execution" +
+                $"{Environment.NewLine}Summary of the errors:" +
+                $"{Environment.NewLine}\t1 instance(s) of: An exception occurred while analyzing a template" +
+                $"{Environment.NewLine}\t1 instance(s) of: Unable to analyze 1 file(s): {Path.Combine(directoryToAnalyze, "ReportsError.json")}" +
+                $"{Environment.NewLine}Summary of the warnings:" +
+                $"{Environment.NewLine}\t1 instance(s) of: An exception occurred when processing the template language expressions{Environment.NewLine}";
+
+            var args = new string[] { "analyze-directory", directoryToAnalyze };
+
+            using StringWriter outputWriter = new();
+            Console.SetOut(outputWriter);
+
+            var result = _commandLineParser.InvokeCommandLineAPIAsync(args);
+
+            var cliConsoleOutput = outputWriter.ToString();
+            var indexOfLogSummary = cliConsoleOutput.IndexOf("2 error(s) and 1 warning(s)");
+            var logSummary = cliConsoleOutput[indexOfLogSummary..];
+
+            Assert.AreEqual(expectedLogSummary, logSummary);
+        }
+
+        [TestMethod]
+        public void AnalyzeDirectory_ExecutionWithErrorAndWarningAndVerboseMode_DoesNotPrintVerboseRecommendation()
```

Comment:
Combine this with the previous one using a `[DataTestMethod]` so the full text is verified with both options.

Reply:
I don't think I'll be able to express something like {Path.Combine(directoryToAnalyze, "ReportsError.json")} as a data test parameter